Ray cluster started. Available resources: {'accelerator_type:G': 1.0, 'CPU': 12.0, 'node:__internal_head__': 1.0, 'node:172.23.48.184': 1.0, 'object_store_memory': 4108806144.0, 'memory': 9587214336.0, 'GPU': 1.0}
Waiting for actors to initialize and run one episode...
[36m(ReplayBufferActor pid=25734)[0m Exception raised in creation task: The actor died because of an error raised in its creation task, [36mray::ReplayBufferActor.__init__()[39m (pid=25734, ip=172.23.48.184, actor_id=f1f7ab6cae702d79c66afa5601000000, repr=<train.ReplayBufferActor object at 0x73f3f01ce710>)
[36m(ReplayBufferActor pid=25734)[0m   File "/home/ryan/toy_mazero/train.py", line 27, in __init__
[36m(ReplayBufferActor pid=25734)[0m     self.replay_buffer = ReplayBuffer(
[36m(ReplayBufferActor pid=25734)[0m   File "/home/ryan/toy_mazero/utils/replay_buffer.py", line 107, in __init__
[36m(ReplayBufferActor pid=25734)[0m     self.observations = np.zeros((capacity, *observation_space), dtype=np.float32)
[36m(ReplayBufferActor pid=25734)[0m TypeError: 'tuple' object cannot be interpreted as an integer
[36m(LearnerActor pid=25728)[0m (Learner pid=25728) Initializing on GPU...
Error in sys.excepthook:
Traceback (most recent call last):
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/linecache.py", line 46, in getlines
    return updatecache(filename, module_globals)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/linecache.py", line 136, in updatecache
    with tokenize.open(fullname) as fp:
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/tokenize.py", line 396, in open
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/tokenize.py", line 365, in detect_encoding
    first = read_or_stop()
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/tokenize.py", line 323, in read_or_stop
    return readline()
KeyboardInterrupt

Original exception was:
Traceback (most recent call last):
  File "/home/ryan/toy_mazero/train.py", line 466, in <module>
    main()
  File "/home/ryan/toy_mazero/train.py", line 391, in main
    ray.get([actor.run_episode.remote() for actor in actors])
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/ray/_private/auto_init_hook.py", line 22, in auto_init_wrapper
    return fn(*args, **kwargs)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/ray/_private/client_mode_hook.py", line 104, in wrapper
    return func(*args, **kwargs)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/ray/_private/worker.py", line 2849, in get
    values, debugger_breakpoint = worker.get_objects(object_refs, timeout=timeout)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/ray/_private/worker.py", line 911, in get_objects
    ] = self.core_worker.get_objects(
  File "python/ray/_raylet.pyx", line 3162, in ray._raylet.CoreWorker.get_objects
  File "python/ray/includes/common.pxi", line 83, in ray._raylet.check_status
KeyboardInterrupt
Exception ignored in atexit callback: <function _start_and_connect_service.<locals>.teardown_atexit at 0x7a7200745a20>
Traceback (most recent call last):
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/lib/service_connection.py", line 90, in teardown_atexit
    conn.teardown(hooks.exit_code)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/lib/service_connection.py", line 218, in teardown
    self._router.join()
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/interface/router.py", line 75, in join
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/threading.py", line 1096, in join
    self._wait_for_tstate_lock()
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/threading.py", line 1116, in _wait_for_tstate_lock
    if lock.acquire(block, timeout):
KeyboardInterrupt:
[0m
Exception ignored in atexit callback: <function shutdown at 0x7a72244e9630>
Traceback (most recent call last):
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/ray/_private/client_mode_hook.py", line 104, in wrapper
    return func(*args, **kwargs)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/ray/_private/worker.py", line 1986, in shutdown
    time.sleep(0.5)
KeyboardInterrupt:
